wordpress-mobile/gutenberg-mobile
# Add editor onboarding capability

* **Gutenberg:** https://github.com/WordPress/gutenberg/pull/31349
* **WPiOS:** https://github.com/wordpress-mobile/WordPress-iOS/pull/16406
* **WPAndroid:** https://github.com/wordpress-mobile/WordPress-Android/pull/14558

Add editor onboarding capability to control visibility of the feature. The flag combines (A) if the user has previously launched the Gutenberg editor and (B) if the user ID falls within the percentage of users allowed. 

This flag is the first task of several to control when the editor onboarding is displayed. This flag will be used in combination with other, future flags (e.g. `tooltipsSeen` tracking which tips have been shown) to determine what should be displayed.

```js
// If the user is eligible to begin onboarding...
if (canViewEditorOnboarding) {
  // Display first tooltip
}

// The user began onboarding, but has not yet seen the second tooltip...
if (tooltipsSeen.length > 1 && !tooltipsSeen.includes('tooltip2')) {
  // Display second tooltip
}
```

To test: 

As this PR and its siblings contain no user-facing changes, the following test plan involves logging out the newly added `capability` to ensure it is working properly. The value behind shipping this addition on its own is to (A) begin tracking editor launches sooner rather than later and (B) produce smaller code diffs to review. 

<details>
<summary>Patch to log editor onboarding flag</summary>

```diff
diff --git a/packages/block-editor/src/components/inserter/index.native.js b/packages/block-editor/src/components/inserter/index.native.js
index 463e03c116..9c3a023466 100644
--- a/packages/block-editor/src/components/inserter/index.native.js
+++ b/packages/block-editor/src/components/inserter/index.native.js
@@ -325,7 +325,12 @@ export default compose( [
 
 			const {
 				__experimentalShouldInsertAtTheTop: shouldInsertAtTheTop,
+				canViewEditorOnboarding,
 			} = getSettings();
+			console.log(
+				'>>> canViewEditorOnboarding',
+				canViewEditorOnboarding
+			);
 
 			// if post title is selected insert as first block
 			if ( shouldInsertAtTheTop ) {
```

</details>

1. Create a patch file on your local system from the diff above.
1. Clone this `gutenberg-mobile` branch. 
1. `cd gutenberg && git apply path/to/patch-file.diff && cd ..`
1. Run the Metro server. 
1. Clone the corresponding branches for WPAndroid and WPiOS, run the main apps leveraging this branch via the integration guides for [WPAndroid](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/WPAndroid%20Integration.md) and [WPiOS](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/WPiOS%20Integration.md) respectively. 
1. Launch the editor. 
1. **Expected:** `>>> canViewEditorOnboarding true` is logged in the Metro server as it is the first time the editor has launched. 
1. Close the editor. 
1. Launch the editor. 
1. **Expected:** `>>> canViewEditorOnboarding false` is logged in the Metro server for all editor launches moving forward. 

PR submission checklist:
